### PR TITLE
Add application insights library for app service monitor

### PR DESCRIPTION
**Pull request by Flanker32**
Add application insights library for app service monitor

### CODE REVIEW

File: azure-toolkit-libs/azure-toolkit-applicationinsights-lib/src/main/java/com/microsoft/azure/toolkit/lib/applicationinsights/ApplicationInsights.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+     * @param name           name for resource to create
+     * @return ApplicationInsightsEntity for created resource
+     */
+    public ApplicationInsightsEntity create(@Nonnull String subscriptionId, @Nonnull String resourceGroup, @Nonnull Region region, @Nonnull String name) {
+        final ApplicationInsightsComponent component = getApplicationInsightsManager(subscriptionId).components().define(name)
+                .withRegion(region.getName()).withExistingResourceGroup(resourceGroup).withKind("web").withApplicationType(ApplicationType.WEB).create();
+        return getFromApplicationInsightsComponent(component);
+    }
+
+    public List<ApplicationInsightsEntity> list() {
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-applicationinsights-lib/src/main/java/com/microsoft/azure/toolkit/lib/applicationinsights/ApplicationInsights.java:78]: eturn Flux.fromIterable(getSubscriptions()).parallel()

File: azure-toolkit-libs/azure-toolkit-applicationinsights-lib/src/main/java/com/microsoft/azure/toolkit/lib/applicationinsights/ApplicationInsightsEntity.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package com.microsoft.azure.toolkit.lib.applicationinsights;
+
+import lombok.Getter;
+import lombok.experimental.SuperBuilder;
+
+@Getter
+@SuperBuilder(toBuilder = true)
+public class ApplicationInsightsEntity {
+    private String subscriptionId;
+    private String resourceId;
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-applicationinsights-lib/src/main/java/com/microsoft/azure/toolkit/lib/applicationinsights/ApplicationInsightsEntity.java:10]: id

File: azure-toolkit-libs/azure-toolkit-applicationinsights-lib/src/main/java/com/microsoft/azure/toolkit/lib/applicationinsights/ApplicationInsightsEntity.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    private String resourceName;
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-applicationinsights-lib/src/main/java/com/microsoft/azure/toolkit/lib/applicationinsights/ApplicationInsightsEntity.java:11]: name

File: azure-toolkit-libs/azure-toolkit-applicationinsights-lib/src/main/java/com/microsoft/azure/toolkit/lib/applicationinsights/ApplicationInsightsEntity.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    private String instrumentationKey;
+    private String type;
+    private String kind;
+    private String location;
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-applicationinsights-lib/src/main/java/com/microsoft/azure/toolkit/lib/applicationinsights/ApplicationInsightsEntity.java:15]: region

File: azure-toolkit-libs/azure-toolkit-applicationinsights-lib/src/main/java/com/microsoft/azure/toolkit/lib/applicationinsights/ApplicationInsights.java (hunk 1 above)
Comment (Flanker32) [reply azure-toolkit-libs/azure-toolkit-applicationinsights-lib/src/main/java/com/microsoft/azure/toolkit/lib/applicationinsights/ApplicationInsights.java:78]: Can not using Flux here as application insights SDK did not provide async interface

**Review by andxu — APPROVED** [review]

**Pull request merged**